File: samples/python/foundry-ai-teammate/src/hello_world_a365_agent/hello_world_a365_agent/models/mcp_server_config.py
"""MCP server configuration models, matching the C# ToolingManifest schema."""

from dataclasses import dataclass, field
from typing import Any


@dataclass
class McpServerConfig:
    mcp_server_name: str = ""
    id: str = ""
    url: str = ""
    scope: str = ""
    audience: str = ""
    publisher: str = ""

    @classmethod
    def from_dict(cls, data: dict[str, Any]) -> "McpServerConfig":
        return cls(
            mcp_server_name=data.get("mcpServerName", ""),
            id=data.get("id", ""),
            url=data.get("url", ""),
            scope=data.get("scope", ""),
            audience=data.get("audience", ""),
            publisher=data.get("publisher", ""),
        )


@dataclass
class ToolingManifest:
    mcp_servers: list[McpServerConfig] = field(default_factory=list)

    @classmethod
    def from_dict(cls, data: dict[str, Any]) -> "ToolingManifest":
        return cls(
            mcp_servers=[McpServerConfig.from_dict(s) for s in data.get("mcpServers", [])]
        )

File: samples/csharp/foundry-workstream-manager-autopilot-agent/src/workstream_manager_agent/Models/McpServerConfig.cs
namespace WorkstreamManager.Models;

using System.Text.Json.Serialization;

public class ToolingManifest
{
    [JsonPropertyName("mcpServers")]
    public List<McpServerConfig> McpServers { get; set; } = [];
}

public class McpServerConfig
{
    [JsonPropertyName("mcpServerName")]
    public string McpServerName { get; set; } = string.Empty;

    [JsonPropertyName("id")]
    public string Id { get; set; } = string.Empty;

    [JsonPropertyName("url")]
    public string Url { get; set; } = string.Empty;

    [JsonPropertyName("scope")]
    public string Scope { get; set; } = string.Empty;

    [JsonPropertyName("audience")]
    public string Audience { get; set; } = string.Empty;

    [JsonPropertyName("publisher")]
    public string Publisher { get; set; } = string.Empty;

    /// <summary>
    /// Optional extra HTTP headers to send to this MCP server alongside the Authorization
    /// bearer. Used by the Foundry toolbox MCP proxy, which requires a feature-flag header
    /// (e.g. "Foundry-Features: Toolboxes=V1Preview") while toolboxes are in preview.
    /// </summary>
    [JsonPropertyName("headers")]
    public Dictionary<string, string>? Headers { get; set; }
}

